datajunction-server/datajunction_server/api/materializations.py

```diff
@@ -286,29 +286,47 @@ async def list_node_materializations(
     """
     Show all materializations configured for the node, with any associated metadata
     like urls from the materialization service, if available.
+
+    show_inactive: bool - Show materializations that have a deactivated_at timestamp set
+    include_all_revisions: bool - Show  materializations for all revisions of the node
     """
     request_headers = dict(request.headers)
-    node = await Node.get_by_name(session, node_name)
+    node = await Node.get_by_name(
+        session,
+        node_name,
+        options=[
+            joinedload(Node.revisions).options(*NodeRevision.default_load_options()),
+        ]
+        if include_all_revisions
+        else [],
+        raise_if_not_exists=True,
+    )
     materializations = []
-    for materialization in node.current.materializations:  # type: ignore
-        if not materialization.deactivated_at or show_deleted:  # pragma: no cover
-            info = query_service_client.get_materialization_info(
-                node_name,
-                node.current.version,  # type: ignore
-                node.type,  # type: ignore
-                materialization.name,  # type: ignore
-                request_headers=request_headers,
-            )
-            if materialization.strategy != MaterializationStrategy.INCREMENTAL_TIME:
-                info.urls = [info.urls[0]]
-            materialization_config_output = MaterializationConfigOutput.from_orm(
-                materialization,
-            )
-            materialization = MaterializationConfigInfoUnified(
-                **materialization_config_output.dict(),
-                **info.dict(),
-            )
-            materializations.append(materialization)
+    if include_all_revisions:  # Get all materializations for all node revisions
+        for node_revision in node.revisions:  # type: ignore
+            for materialization in node_revision.materializations:
+                if (
+                    not materialization.deactivated_at or show_inactive
+                ):  # pragma: no cover
```

Comment:
I think there is coverage for this line right?

Reply:
You're right, I'm not sure why I put this here. 🤔